File: baekjoon/1924.c
#include<stdio.h>

int DAY(int m);
int main()
{
    int x, y;
    scanf("%d %d", &x, &y);

    x--;
    while (x >= 1)
    {
        y += DAY(x);
        x--;
    }
    switch (y % 7)
    {
    case 1: printf("MON");
        break;
    case 2: printf("TUE");
        break;
    case 3: printf("WED");
        break;
    case 4: printf("THU");
        break;
    case 5: printf("FRI");
        break;
    case 6: printf("SAT");
        break;
    case 0: printf("SUN");
        break;
    }
    return 0;
}

int DAY(int m)
{
    int day;
    if (m == 1 || m == 3 || m == 5 || m == 7 || m == 8 || m == 10 || m == 12)
        day = 31;
    else if (m == 2)
        day = 28;
    else
        day = 30;

    return day;
}

File: baekjoon/1000/1924.rs
use std::io::*;

struct Scanner {
    it: std::str::SplitAsciiWhitespace<'static>,
}

impl Scanner {
    fn new() -> Self {
        let mut s = String::new();
        stdin().read_to_string(&mut s).ok();
        Self { it: s.leak().split_ascii_whitespace() }
    }
    fn read<T: std::str::FromStr>(&mut self) -> T {
        self.it.next().unwrap().parse::<T>().ok().unwrap()
    }
}

fn main() {
    let mut sc = Scanner::new();
    let mut bw = BufWriter::new(stdout().lock());
    const DAYS: [usize; 12] = [31, 28, 31, 30, 31, 30, 31, 31, 30, 31, 30, 31];
    const DAY: [&str; 7] = ["MON", "TUE", "WED", "THU", "FRI", "SAT", "SUN"];
    let m = sc.read::<usize>();
    let d = sc.read::<usize>() + DAYS.iter().take(m - 1).sum::<usize>();
    writeln!(bw, "{}", DAY[(d - 1) % 7]).ok();
}
